server/tomlconfig/verify.go

```diff
@@ -132,6 +134,28 @@ func (c *SXGCertConfig) verify() error {
 	return errs.ErrorOrNil()
 }
 
+func (c *SXGACMEConfig) verify() error {
+	var errs *multierror.Error
+
+	if !c.EnableCertAutoRenew {
+		return errs.ErrorOrNil()
+	}
+
+	if c.DiscoveryURL == "" {
+		errs = multierror.Append(errs, wrapError("DiscoveryURL", errEmpty))
+	}
+	if c.Email == "" {
+		errs = multierror.Append(errs, wrapError("Email", errEmpty))
+	}
+
+	if c.HTTPWebRootDir == "" && c.HTTPChallengePort == 0 &&
+		c.TLSChallengePort == 0 && c.DNSProvider == "" {
+		errs = multierror.Append(errs, wrapError("at least one of HTTPWebRootDir, HTTPChallengePort, TLSChallengePort or DNSProvider has to be non-empty", errEmpty))
```

Comment:
It isn't required to have exactly one parameter. They can all be set and it's up to the library to decide which method to use and to use the other settings as backup.

> The first argument of wrapError is meant to be the parameter names, so it should be something like one of {HTTP..., HTTP..., TLS..., DNS...}.

So I'm not sure how this would work when we are generating the error because none of these parameters have values.  Do I pass in all the parameter names with empty values to wrapError, is that what you mean?

Reply:
I think you get it mostly right, I meant the first argument of `wrapError` should look like a parameter name (a list of parameter names in this case) rather than a sentence. With another thought, I suggest:

```go
newError(
    "HTTPWebRootDir, HTTPChallengePort, TLSChallengePort, DNSProvider",
    "at least one of these parameters must be non-empty",
)
```

Technically the parameter name(s) are just printed as part of error messages ([Code](https://github.com/google/webpackager/blob/toml/server/tomlconfig/error.go#L38)).

> It isn't required to have exactly one parameter. They can all be set and it's up to the library to decide which method to use and to use the other settings as backup.

This makes sense, but it's worth mentioned in the example toml.

Reply:
Done.